content/what-is-cross-origin-resource-sharing/index.md

```diff
@@ -0,0 +1,166 @@
+
+Now let's take a look at the response from the server.
+
+```
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: https://emailpassword.demo.supertokens.com/
+Access-Control-Allow-Credentials: truefront-token: ...
+Access-Control-Expose-Headers: front-token, id-refresh-token
+Set-Cookie: ...
+```
+
+Here, we get the response from the server with cookies and tokens that allows us to proceed with the sign-in operation. One thing to note is that compared to our pre-flight, we now also have an additional `Access-Control-Expose-Headers` header.
+
+- `Access-Control-Expose-Headers` - The server indicating which response headers are available to scripts running in the browser.
+
+With this, we've now completed our first pre-flight request/response as well as our actual request/response for signing in!
+
+## Wildcards in CORS
+
+One common mistake in configuring CORS is around the use of wildcards. Often, developers will elect to use the wildcard, `*`, when defining the origins, methods, or headers allowed with CORS.
+
+While wildcards will work for simple requests (requests without [HTTP cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) or HTTP authentication information), requests with credentials will often encounter a CORS not authorized error.
+
+That's because in requests with credentials (cookies), the wildcard is treated as the literal method name or origin name "*" without special semantics. This occurs both in `Access-Control-Allow-Origin` and `Access-Control-Allow-Methods`. And some browsers like Safari simply don't have support for wildcards at all, simple request or not.
+
+All in all, it's good hygiene to avoid the wildcard and use a comma-separated list when configuring CORS.
+
+## CORS vulnerability
+
+When configured improperly, CORS can also lead to major vulnerabilities. Below, we'll list a couple of common issues when configuring CORS.
+
+**Mishandling origin whitelist**
+
+One of the easiest mistakes to make when implementing CORS is mishandling the origin whitelist. When whitelisting origins, it's often easy to do simple matches with URL prefixes or suffixes, or using regular expressions. However this can lead to quite a few issues. Let's look at some examples below.
+
+Let's say that we grant access to all websites with the prefix `whitelisted-website.com`. This helps us grant access to origins such as `whitelisted-website.com/blog` or `whitelisted-website.com/login`.
+
+But a malicious actor might come around and use the origin - `whitelisted-website.com.malicious.com` and bypass our security measures.
+
+Another approach that we could use is to grant all websites with the suffix `whitelisted-website.com`. This makes it easy for us to grant access to `api.whitelisted-website.com`.
```

Comment:
```suggestion
Another approach that we could use is to grant all websites with the suffix `whitelisted-website.com`. This makes it easy for us to grant access to `user1.whitelisted-website.com`.
```

Reply:
Using api sub domain as an example is wrong cause api sub domain won't be something thats navigated to on a browser.